substrate/frame/tips/src/tests.rs

```diff
@@ -654,3 +654,15 @@ fn report_awesome_and_tip_works_second_instance() {
 		assert_eq!(Balances::free_balance(&Treasury1::account_id()), 191);
 	});
 }
+
+#[test]
+fn try_state_invariant_works() {
+	new_test_ext().execute_with(|| {
+		use frame_support::pallet_prelude::DispatchError::Other;
+		Balances::make_free_balance_be(&Treasury::account_id(), 101);
+
+		assert_ok!(Tips::report_awesome(RuntimeOrigin::signed(0), b"".to_vec(), 3));
+
+		assert_eq!(Tips::do_try_state(), Err(Other("finder's fee not present")))
```

Comment:
w.r.t my comment above you would need to fix it here as well, i.e.
```suggestion
		assert_eq!(Tips::do_try_state(), Err(Other("Tips with `finders_fee` should have non-zero `deposit`.")))
```

Reply:
done 🤝